### DISCUSSION TITLE
Support statblock adjustment rules from systems like e.g. Pathfinder 2E

### DISCUSSION THREAD

**Discussion by JamesYeoman** [Obsidian-TTRPG-Community/fantasy-statblocks#334] [Feature Requests]
The adjustment rules can be found [here on Archives Of Nethys](https://2e.aonprd.com/Rules.aspx?ID=788), and all the creature pages have the variants built-in (see [the Elite Weretiger](https://2e.aonprd.com/Monsters.aspx?ID=858&Elite=true), the variant links have query parameters to indicate the variant).

Not quite sure how this would be implemented. Maybe it could be implemented as a custom layout? Or maybe it needs to be implemented as a JS layer over the top, with a radio button to select the variant?

**Comment by ThaddaeusWiedemer** [comment]
If I understand your goal correctly (I'm not familiar with PF2E), I think you could simply create a copy of the PF2E layout, and add a JavaScript block in the very beginning. In the block, you can define all the rules, like e.g.

```javascript
function eliteAdjustment () {
  monster.ac = monster.ac + 2;
  if (monster.level <= 1) {
    monster.level = monster.level + 10;
  } else if (monster.level <= 4) {
    // ..
  }
  // other adjustments
}
```

Some of the rules would probably require you to parse e.g. the `monster.skills` property to first retrieve the values, then modify them, then put them back into the string, but it should be doable.

In the same JavaScript Block you could programmatically create some radio buttons for each variant.

The block would then modify the data based on your implemented rules and everything else would be rendered as usual.
